identity/src/main/kotlin/net/thechance/identity/service/ResetPasswordService.kt

```diff
@@ -0,0 +1,82 @@
+package net.thechance.identity.service
+
+import net.thechance.identity.api.dto.RequestOtpResponse
+import net.thechance.identity.api.dto.VerifyOtpResponse
+import net.thechance.identity.entity.OtpLog
+import net.thechance.identity.exception.*
+import net.thechance.identity.repository.OtpLogRepository
+import net.thechance.identity.repository.UserRepository
+import net.thechance.identity.service.otpGenerator.OtpGeneratorService
+import net.thechance.identity.service.phoneNumberValidator.PhoneNumberValidatorService
+import net.thechance.identity.service.sms.SmsService
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
+import java.time.Instant
+import java.util.UUID
+
+@Service
+class ResetPasswordService(
+    private val phoneNumberValidatorService: PhoneNumberValidatorService,
+    private val otpGeneratorService: OtpGeneratorService,
+    private val smsService: SmsService,
+    private val otpLogRepository: OtpLogRepository,
+    private val userRepository: UserRepository
+) {
+    fun requestOtp(phoneNumber: String, defaultRegion: String): RequestOtpResponse {
+        val validatedPhoneNumber = phoneNumberValidatorService.validateAndParse(phoneNumber, defaultRegion)
+        checkPhoneNumberExistence(phoneNumber)
+        checkRequestLimitByPhoneNumber(phoneNumber)
+        expireOldActiveOtpByPhoneNumber(phoneNumber)
+        val otp = otpGeneratorService.generateOtp()
+        val otpLog = OtpLog(
+            phoneNumber = phoneNumber,
+            otp = otp,
+            expireAt = Instant.now().plusSeconds(OTP_EXPIRY_SECONDS),
+        )
+        otpLogRepository.save(otpLog)
+        smsService.sendSms(
+            validatedPhoneNumber.countryCode,
+            validatedPhoneNumber.carrierPrefixHeuristic,
+            validatedPhoneNumber.phoneNumber,
+            otp,
+        )
+        return RequestOtpResponse(otpLog.sessionId.toString())
+    }
+
+    fun verifyOtp(otp: String, sessionId: String): VerifyOtpResponse {
+        val parsedSessionId = UUID.fromString(sessionId)
+        val otpLog = otpLogRepository.findByOtpAndSessionId(otp, parsedSessionId)
+            ?: throw InvalidOtpException()
+        if (otpLog.isVerified) throw InvalidOtpException()
+        if (otpLog.expireAt.isBefore(Instant.now())) throw OtpExpiredException()
+        otpLogRepository.verifyOtp(parsedSessionId)
+        return VerifyOtpResponse("OTP verified successfully")
+    }
+
+    private fun checkPhoneNumberExistence(phoneNumber: String) {
+        userRepository.findByPhoneNumber(phoneNumber) ?: throw UserNotFoundException("User not found")
+    }
+
+    private fun expireOldActiveOtpByPhoneNumber(phoneNumber: String) {
+        otpLogRepository.updateExpirationByPhoneNumber(phoneNumber)
+    }
+
+    private fun checkRequestLimitByPhoneNumber(phoneNumber: String) {
```

Comment:
this is not the responsibility of ResetPasswordService I suggest extracte it into separate class like: PhoneNumberLimiter